packages/block-editor/src/components/media-replace-flow/index.js

```diff
@@ -0,0 +1,231 @@
+			// Stop the key event from propagating up to ObserveTyping.startTypingInTextField.
+			event.stopPropagation();
+		}
+	};
+
+	const selectMedia = ( media ) => {
+		onSelect( media );
+		setMediaURLValue( media.url );
+		setShowMediaReplaceOptions( false );
+		speak( __( 'The media file has been replaced' ) );
+	};
+
+	const onUploadError = ( message ) => {
+		noticeOperations.removeAllNotices();
+		noticeOperations.createErrorNotice( message );
+	};
+
+	const uploadFiles = ( event ) => {
+		const files = event.target.files;
+		let setMedia;
+		if ( multiple ) {
+			setMedia = selectMedia;
+		} else {
+			setMedia = ( [ media ] ) => {
+				setShowMediaReplaceOptions( false );
+				selectMedia( media );
+			};
+		}
+		mediaUpload( {
+			allowedTypes,
+			filesList: files,
+			onFileChange: setMedia,
+			onUploadError,
+		} );
+	};
+
+	const fileUploadButton = (
+		<FormFileUpload
+			onChange={ uploadFiles }
+			accept={ allowedTypes }
+			multiple={ multiple }
+			render={ ( { openFileDialog } ) => {
+				return (
+					<MenuItem
+						icon="upload"
+						onClick={ () => {
+							openFileDialog();
+						} }
+					>
+						{ __( 'Upload' ) }
+					</MenuItem>
+				);
+			} }
+		/>
+	);
+
+	const URLButton = (
+		<MenuItem
+			icon="admin-links"
+			onClick={ () => ( setShowURLInput( ! showURLInput ) ) }
+			aria-expanded={ showURLInput }
+		>
+			<div> { __( 'Insert from URL' ) } </div>
+		</MenuItem>
+	);
+
+	let urlInputUIContent;
+	if ( showEditURLInput ) {
+		urlInputUIContent = ( <LinkEditor
+			onKeyDown={ stopPropagationRelevantKeys }
+			onKeyPress={ stopPropagation }
+			value={ mediaURLValue }
+			isFullWidthInput={ true }
+			hasInputBorder={ true }
+			onChangeInputValue={ ( url ) => ( setMediaURLValue( url ) ) }
+			onSubmit={ ( event ) => {
+				event.preventDefault();
+				onSelectURL( mediaURLValue );
+				setShowEditURLInput( ! showEditURLInput );
+				editMediaButtonRef.current.focus();
+			} }
+		/> );
+	} else {
+		urlInputUIContent = ( <LinkViewer
+			isFullWidth={ true }
+			className="editor-format-toolbar__link-container-content block-editor-format-toolbar__link-container-content"
+			url={ mediaURLValue }
+			onEditLinkClick={ () => ( setShowEditURLInput( ! showEditURLInput ) ) }
+		/> );
+	}
+
+	const urlInputUI = ( showURLInput && <div className="block-editor-media-flow__url-input">
+		{ urlInputUIContent }
+	</div> );
+
+	const onClose = () => {
+		editMediaButtonRef.current.focus();
+	};
+
+	const onClickOutside = () => {
+		return ( ) => {
+			setShowMediaReplaceOptions( false );
+		};
+	};
+
+	const editMediaButton = (
+		<MediaUploadCheck>
+			<MediaUpload
+				onSelect={ ( media ) => selectMedia( media ) }
+				onClose={ () => setShowMediaReplaceOptions( true ) }
+				allowedTypes={ allowedTypes }
+				render={ ( { open } ) => (
+					<>
+						<Toolbar className={ 'mediaflow-replace components-dropdown-menu components-toolbar' }>
```

Comment:
Why does this repeat the classes that should come through the component itself? Also, the component is called `media-replace-flow` and this class is `mediaflow-replace`.

Reply:
> the component is called media-replace-flow and this class is mediaflow-replace.
Yeah, this is because of all the refactoring/renaming that has been going on :) Will revert all to `media-replace-flow`

> Why does this repeat the classes that should come through the component itself? 
`components-toolbar` is a mistake (copypasta)
`components-dropdown-menu` is a trick b/c I didn't want to repeat the CSS since this whole component is a dropdown in disguise.